### ISSUE TITLE
IoTHubMessage_SetMessageCreationTimeUtcSystemProperty  doesnt add iothub-creation-time-utc in system property.

### ISSUE THREAD

**Issue by rajaggrawal** [Azure/azure-iot-sdk-c#1970]
This is follow-up about following issue.
https://github.com/Azure/azure-iot-sdk-c/issues/1568.

IoTHubMessage_SetMessageCreationTimeUtcSystemProperty is adding **iothub-creation-time-utc** in application-property but it should be a system property.

Note :
We already had method(IoTHubMessage_SetProperty) to add application-property and i can set **iothub-creation-time-utc** property using this method.

**Comment by ericwol-msft** [comment]
The documentation shows this as an application property.

https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-construct

**Comment by rajaggrawal** [comment]
@ericwol-msft  Thanks for update.
Yes its correct as per documentation.

But method name (IoTHubMessage_SetMessageCreationTimeUtcSystemProperty) itself says its a system property :)
Also if this is a application property then we really may not need of IoTHubMessage_SetMessageCreationTimeUtcSystemProperty, as application itself can add this property(key, value) without this method.

**Comment by ericwol-msft** [comment]
There is two views here, a device centric and an application centric view.

**Comment by ericwol-msft** [comment]
There is two views here, a device centric and an application centric view.

**device centric**
From the device, this is a reserved system property defined by the hub. For MQTT messages, this value is set in the message topic using the reserved ".ctime" [property name](https://github.com/Azure/azure-iot-sdk-java/blob/13c32cc00b57532bcb29f204199b122ea06f20a9/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/Mqtt.java#L55). Any property name that started with a period is a reserved system property.

devices/snoopy2/messages/events/.ce=utf-8&%24 **.ctime=2020-07-01T01%3a00%3a00.346Z**

From the C SDK perspective, this naming convention is inline with the [C# SDK](https://github.com/Azure/azure-iot-sdk-csharp/blob/cd93e75fd8914b6247f23eb6a5cfd7aea459676f/iothub/device/src/MessageSystemPropertyNames.cs) as well.

**application centric**

Telemetry messages are usually received by the AMQP protocol on the event hub. AMQP messages has "system properties" that are predefined properties names. Where as possible, the hub system message properties are mapped to the AMQP message properties, for example 'content_encoding'. For hub system properties that don't map directly to the AMQP message "system properties", then the AMQP protocol requires you to use the AMQP "application properties". Since there is no predefined AMQP property name for message creation time, a property value in the AMQP application properties is created to store the value.

**TL;DR** System properties are protocol specific. The IoT Hub reserves certain property names as message system properties.

@ewertons

**Comment by danewalton** [comment]
I am going to go ahead and close this issue for now for inactivity. Please let us know if you would like it reopened.
